src/studio/src/designer/frontend/packages/schema-editor/src/utils.ts

```diff
@@ -167,23 +208,61 @@ export function buildUISchema(schema: any, rootPath: string, includeDisplayName:
   return result;
 }
 
+export const mapGroupItemToUiSchemaItem = (item: any, index: number, key: GroupKeys, parentPath: string) => {
+  return {
+    ...item,
+    path: `${parentPath}/${key}/${index}`,
+    displayName: item.$ref !== undefined ? 'ref' : 'Inline object',
+    groupItem: true,
+  };
+};
+
+export const nullType = (item: UiSchemaItem) => item.type?.toLowerCase() === 'null';
+
+export const groupIsNullable = (item: UiSchemaItem): boolean => {
+  return Boolean(
+    item?.anyOf?.some(nullType) ||
+    item?.allOf?.some(nullType) ||
+    item?.oneOf?.some(nullType),
+  );
+};
+
+export const mapGroupChildren = (children: UiSchemaItem[], toType: GroupKeys): UiSchemaItem[] => {
+  // example case, going from oneOf to allOf
+  // example input: #/definitions/allOfTest/allOf/1/something/properties/oneOfTest/oneOf/0
+  // example output: #/definitions/allOfTest/allOf/1/something/properties/oneOfTest/allOf/0
+  return children.map((child) => {
```

Comment:
these are always subSchemas? Maybe a more fitting name?

Reply:
Well, they can be any UiSchemaItem, a "group" can have sub groups and so on. I think children is fine for now.